src/core/detail/bitpacking.cc

```diff
@@ -104,6 +108,33 @@ static inline pair<const char*, uint8_t*> simd_variant2_pack(const char* ascii,
 // See https://github.com/lemire/fastvalidate-utf-8/
 // The function returns true (1) if all chars passed in src are
 // 7-bit values (0x00..0x7F). Otherwise, it returns false (0).
+#ifdef __s390x__
+bool validate_ascii_fast(const char* src, size_t len) {
+  size_t i = 0;
+  vector unsigned char has_error = vec_splat_s8(0);
+  if (len >= 16) {
+    for (; i <= len - 16; i += 16) {
+      vector unsigned char current_bytes = vec_load_len((signed char*)(src + i), 16);
+      has_error = vec_orc(has_error, current_bytes);
+    }
+  }
+
+  int error_mask = 0;
+  for (int i = 0; i < 16; i++) {
+    if (has_error[i]) {
+      error_mask |= 1 << i;
+    }
+  }
+
+  char tail_has_error = 0;
+  for (; i < len; i++) {
+    tail_has_error |= src[i];
+  }
+  error_mask |= (tail_has_error & 0x80);
```

Comment:
what error_mask holds?

Reply:
The variable 'error_mask' stores the final product of the OR operations aligned by 128-bit buffers.
The variable should be less than 0x80 which annotates that the string is a valid ASCII string.